packages/iTwinUI-react/src/core/Table/TableRowMemoized.tsx

```diff
@@ -49,8 +54,30 @@ export const TableRow = <T extends Record<string, unknown>>(props: {
     isLast && onBottomReached.current?.();
   }, [isLast, onBottomReached, onRowInViewport, row.original]);
 
-  const rowRef = useIntersection(onIntersect, {
+  const [rowRef, setRowRef] = React.useState<HTMLElement | null>();
+  const getIntersectionRoot = () => {
+    const isTableBodyScrollable =
+      (rowRef?.parentElement?.scrollHeight ?? 0) >
+      (rowRef?.parentElement?.offsetHeight ?? 0);
+    // If table body is scrollable, make it the intersection root
+    if (isTableBodyScrollable) {
+      return rowRef?.parentElement;
+    }
+
+    const scrollableParent = getScrollableParent(
+      rowRef?.closest('.iui-table'),
+      rowRef?.ownerDocument,
+    );
+    // If there is no scrollable parent besides body, then return undefined to make viewport as a root
+    if (scrollableParent === rowRef?.ownerDocument.body) {
+      return undefined;
+    }
+    return scrollableParent;
```

Comment:
Do we ever make the table scrollable but not the table body? Feels like we can just leave it as before if it's not table body.

```suggestion
    return undefined;
```

Reply:
Maybe it doesn't make sense if users have Table inside some div, idk.

Reply:
Even if it's inside a div, we still make only the table body scrollable, right?

Plus I feel like it makes sense that if you want to use our built in intersectionMargin that you need to use our scroll method. And if you control the scrolling outside the table, you should bring your own intersection observer.

Reply:
No, only if the table has something like `height: 100%` because else scroll will go on the wrapper div. It is kind of the same as in Storybook some Table stories.

Reply:
Simplified things, if there will be a need, we can add that handling later.